platforms/core-configuration/configuration-cache/src/test/kotlin/org/gradle/internal/cc/impl/problems/ConfigurationCacheProblemsSummaryTest.kt

```diff
@@ -268,32 +270,87 @@ See the complete report at $REPORT_URL
         val subject = ConfigurationCacheProblemsSummary()
         subject.onProblem(buildLogicProblem(buildLogicLocationTrace("build.gradle.kts", 1), "failure"), ProblemSeverity.Deferred)
         subject.onProblem(buildLogicProblem(buildLogicLocationTrace("build.gradle.kts", 1), "failure"), ProblemSeverity.Deferred)
-        assertThat(subject.get().uniqueProblemCount, equalTo(1))
+        assertThat(subject.get().consoleUniqueProblemCount, equalTo(1))
 
         subject.onProblem(buildLogicProblem(buildLogicLocationTrace("build.gradle.kts", 2), "failure"), ProblemSeverity.Deferred)
-        assertThat(subject.get().uniqueProblemCount, equalTo(2))
+        assertThat(subject.get().consoleUniqueProblemCount, equalTo(2))
+    }
+
+    @Test
+    fun `problem causes are as expected`() {
+        val gradleRuntime = ProblemCause.of(PropertyProblem(PropertyTrace.Gradle, StructuredMessage.forText("failure")))
+        assertThat(gradleRuntime.message, equalTo("failure"))
+        assertThat(gradleRuntime.userCodeLocation, equalTo("Gradle runtime"))
+
+        val unknown = ProblemCause.of(PropertyProblem(PropertyTrace.Unknown, StructuredMessage.forText("failure")))
+        assertThat(unknown.message, equalTo("failure"))
+        assertThat(unknown.userCodeLocation, equalTo("unknown location"))
+
+        val projectAtUnknown = ProblemCause.of(PropertyProblem(PropertyTrace.Project(":p1", PropertyTrace.Unknown), StructuredMessage.forText("failure")))
+        assertThat(projectAtUnknown.message, equalTo("failure"))
+        assertThat(projectAtUnknown.userCodeLocation, equalTo("unknown location"))
```

Comment:
🤔 I might be wrong, but we're testing here just values to be set. How useful this test is?